c-dependencies/js-compute-runtime/Makefile

```diff
@@ -53,8 +58,33 @@ $(RUST_URL_LIB): $(RUST_URL_RS_FILES) $(RUST_URL_SRC)/Cargo.toml $(RUST_URL_SRC)
 	$(WASI_CXX) $(CXX_FLAGS) $(CXX_OPT) $(DEFINES) -I $(SM_SRC)include -MMD -MP -c -o $@ $<
 
 js-compute-runtime.wasm: $(FSM_OBJ) $(SM_OBJ) $(RUST_URL_LIB)
-	$(WASI_CXX) $(CXX_FLAGS) $(CXX_OPT) $(DEFINES) $(LD_FLAGS) -o $@ $^
-	$(WASM_STRIP)
+	$(WASI_CXX) $(CXX_FLAGS) $(CXX_OPT) $(DEFINES) $(LD_FLAGS) -o $@ $^ && \
+		$(call WASM_STRIP,$@)
 
 initialized-js-compute-runtime.wasm: js-compute-runtime.wasm $(INIT_JS)
 	cat $(INIT_JS) | $(WIZER) --allow-wasi --dir=. -r _start=wizer.resume -o $@ $<
+
+clean:
+	$(RM) compile_commands.json $(FSM_OBJ)
+
+distclean: clean
+	$(RM) $(FSM_DEP) compiler_flags
+
+# technically this depends on the value of the FSM_CPP variable
+compile_commands.json: compiler_flags
+	$Q ( \
+		sep="["; \
+		for file in $(FSM_CPP); do \
+			echo "$$sep"; \
+			sep=","; \
+			echo "{ \"directory\": \"$(FSM_SRC)\","; \
+			echo "  \"command\": \"$(WASI_CXX) $(CXX_FLAGS) $(DEFINES) -I $(SM_SRC)include\","; \
+			echo -n "  \"file\": \"$$(basename $$file)\"}"; \
+		done; \
+		echo; \
+		echo ']' \
+	) > "$@"
```

Comment:
All I know about `compile_commands.json` is that when I tried to generate one once I got confused and gave up. Is this the best way to do it? I thought I remembered clang having an option to generate it for you, maybe?

Reply:
If clang can do it automatically, that would be better. This version doesn't feel ideal, but does produce a working compile_commands.json :)

Reply:
A quick googling didn't turn up anything that automatically generates compile_commands.json, and my experience with compile_commands.json from bazel and cmake is that it's generated by the build system, not clang/clangd. How do you feel about using this hacky version for now, and replacing it with something more robust when we find it?

Reply:
_If_ we find it. It's entirely possible I gave up the last time because there isn't a nice option if your build system doesn't do it for you; I don't remember. Yeah, I'm satisfied, ship it!

Maybe we should discuss changing the build system here to something more capable. That's certainly not blocking this PR though.